Dockerfile

```diff
@@ -0,0 +1,35 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file builds the Linux-x86_64, Linux-arm, Linux-armfh. Linux aarch64 and Win64 jnilibs.  It copies the contents of
+# the build host's project directory (commons-crypto) into the docker image and cross compiles the remaining builds. If 
+# you run this script from a Mac after a successful build, the build in the resuing Docker image will include the Mac 
+# build.
+
+FROM ubuntu:18.04
```

Comment:
For building binaries to be used in a release, this should be ubuntu:14.04 or even centos:7 to retain as much compatibility as possible.

Reply:
If we want to simplify the docker files to not cross-compile, we could also add this to the FROM clause: `--platform=linux/aarch64`

Reply:
14.04 is end of life, so I went with the most current LTS version of Ubuntu.  I chose Ubuntu because the comments in the makefile specified the proper packages for cross-compiling.  I opted for the cross-compile approach as opposed to pulling down a new base image for each platform so that each iteration would add an architecture to the existing build.  If I built each architecture on a separate base image I'd have to do a lot of build-dir copying between images.

Reply:
Sure, though for purposes of a release, it'd be best to take a distro that is 'old'.  The main concern is libc compatibility as compiling against newer versions **is not** backward compatible to older versions, whereas compiling against an old version of libc **is** forward compatible.  

Ubuntu 14.04 is still supported by Canonical according to their Docker hub info page, and is old enough for Arm builds, but for x86, it'd be better to go further back to something like CentOS7 or even 6.

Reply:
FWIW, what I have working now for my local builds is based on 14.04 which is what I'll use to cut a release candidate hopefully soon(ish).